device/Microsoft.Azure.Devices.Client/ModuleAuthenticationWithTokenRefresh.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    using System;
+    using Microsoft.Azure.Devices.Client.Extensions;
+
+    /// <summary>
+    /// Authentication method that uses a shared access signature token and allows for token refresh. 
+    /// </summary>
+    public abstract class ModuleAuthenticationWithTokenRefresh : DeviceAuthenticationWithTokenRefresh
```

Comment:
Would simply adding "ModuleId" to the current type achieve the same functionality?

Reply:
It would... it is just now what we have done with other AuthenticationMethod types...

Reply:
We should try to limit code duplication as much as possible: it could be that a different abstract can be extracted between this and the `Device` equivalent.